src/app/app.js

```diff
@@ -0,0 +1,427 @@
+
+                if (window.state !== route.state) {
+                    $('#preloader').show();
+                    $('body').removeClass('hidden');
+
+                    $('head link').each(function (index, element) {
+                        element.remove();
+                    });
+
+                    for (
+                        var index = 0;
+                        index < route.css.length;
+                        index++
+                    ) {
+                        $('head').prepend(
+                            (
+                                '<link rel="stylesheet" href="' +
+                                Object.assign([], route.css).reverse()[index] +
+                                '">'
+                            )
+                        );
+                    }
+
+                    toggleAppElements(route.state === 'app');
+                    toggleDashboardElements(route.state === 'dashboard');
+
+                    $('#app').empty();
+                    $('#dashboard').empty();
+                }
+
+                setActiveNavItem(
+                    match.route.path,
+                    $(route.menuItems)
+                );
+
+                function loadPage() {
+                    $(route.selector).load(
+                        route.template,
+                        function () {
+                            $.ajax({
+                                url: route.controller,
+                                dataType: 'script'
+                            });
+                            window.router.updatePageLinks();
+                        }
+                    );
+                    done();
+                }
+
+                function success(currentUserResponse) {
+                    if ($.inArray(match.route.path, ['signin', 'signup']) !== -1) {
+                        window.router.navigate('/');
+                        done(false);
+                    } else {
+                        window.user = currentUserResponse;
+                        $('.user-name').text(window.user.name);
+
+                        toggleAuthMenu(false);
+                        loadPage();
+                    }
+                }
+
+                function error() {
+                    if (route.state === 'dashboard') {
+                        window.router.navigate('/');
+                        done(false);
+                    } else {
+                        toggleAuthMenu(true);
+                        loadPage();
+                    }
+                }
+
+                $.ajax({
+                    type: 'GET',
+                    url: window.config.api + 'api/current_user/',
+                    xhrFields: {withCredentials: true},
+                    success: success,
+                    error: error,
+                });
+            }
+
+            function after(match) {
+                window.state = window.routes[match.route.path].state;
+            }
+
```

Comment:
Done

Reply:
Remove above code from `auth.js`, it's still there, but not necessary since you have it now in `app.js`

Reply:
Still there 

![mroomapp – auth js 2022-01-29 17-18-51](https://user-images.githubusercontent.com/12160621/151666445-08ea9c43-afc8-48d2-af35-0b096cce2251.png)

Reply:
You shouldn't keep `$(window).scroll` inside `function before(done, match)` @MirabbosMirzakhmedov 

Move it outside of `function before(done, match)`, you can keep it right after `window.getFormData`

![mroomapp – app js 2022-02-08 13-08-59](https://user-images.githubusercontent.com/12160621/152975663-e388344e-0177-4d90-a1ad-cd3a0546603c.png)

Reply:
If you keep `$(window).scroll` inside `before` function, scroll event will be defined every time you change the page - see when `before` function is called.

This can cause a lot of unexpected situations, glitches, etc.

Reply:
Resolved